tests/fixture/tmpnet/network.go

```diff
@@ -781,6 +796,41 @@ func (n *Network) getBootstrapIPsAndIDs(skippedNode *Node) ([]string, []string,
 	return bootstrapIPs, bootstrapIDs, nil
 }
 
+// Waits until the provided nodes are healthy.
+func waitForHealthy(ctx context.Context, w io.Writer, nodes []*Node) error {
+	ticker := time.NewTicker(networkHealthCheckInterval)
+	defer ticker.Stop()
+
+	healthyNodes := set.NewSet[ids.NodeID](len(nodes))
+	for healthyNodes.Len() < len(nodes) {
+		for _, node := range nodes {
+			if healthyNodes.Contains(node.NodeID) {
+				continue
+			}
+
+			healthy, err := node.IsHealthy(ctx)
+			if err != nil && !errors.Is(err, ErrNotRunning) {
+				return err
+			}
+			if !healthy {
+				continue
+			}
+
+			healthyNodes.Add(node.NodeID)
+			if _, err := fmt.Fprintf(w, "%s is healthy @ %s\n", node.NodeID, node.URI); err != nil {
+				return err
+			}
+		}
+
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("failed to see all nodes healthy before timeout: %w", ctx.Err())
+		case <-ticker.C:
+		}
+	}
+	return nil
+}
```

Comment:
The only actual change here is that this this reduces the time this function takes to return by `networkHealthCheckInterval`.

I also felt it was a bit cleaner to not iterate over nodes that were already healthy... but 🤷

 
```suggestion
// Waits until the provided nodes are healthy.
func waitForHealthy(ctx context.Context, w io.Writer, nodes []*Node) error {
	ticker := time.NewTicker(networkHealthCheckInterval)
	defer ticker.Stop()

	unhealthyNodes := set.Of(nodes...)
	for {
		for node := range unhealthyNodes {
			healthy, err := node.IsHealthy(ctx)
			if err != nil && !errors.Is(err, ErrNotRunning) {
				return err
			}
			if !healthy {
				continue
			}

			unhealthyNodes.Remove(node)
			if _, err := fmt.Fprintf(w, "%s is healthy @ %s\n", node.NodeID, node.URI); err != nil {
				return err
			}
		}

		if unhealthyNodes.Len() == 0 {
			return nil
		}

		select {
		case <-ctx.Done():
			return fmt.Errorf("failed to see all nodes healthy before timeout: %w", ctx.Err())
		case <-ticker.C:
		}
	}
}
```